File: dev-superior-algoritmo-e-logica-de-programacao/c/exercicios/maior_posicao.c
#include <stdio.h>

int main() {
    int N, posicao;
    double maiorValor;

    printf("Quantos numeros voce vai digitar? ");
    scanf("%d", &N);

    double vet[N];

    for (int i = 0; i < N; i++) {
        printf("Digite um numero: ");
        scanf("%lf", &vet[i]);
    }

    posicao = 0;
    maiorValor = vet[0];

    for (int i = 1; i < N; i++) {
        if (vet[i] > maiorValor) {
            maiorValor = vet[i];
            posicao = i;
        }
    }

    printf("\nMAIOR VALOR = %.1lf\n", maiorValor);
    printf("POSICAO DO MAIOR VALOR = %d\n", posicao);

    return 0;
}

File: dev-superior-algoritmo-e-logica-de-programacao/java/src/exercicios/MaiorPosicao.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class MaiorPosicao {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, posicao;
		double maiorValor;

	    System.out.print("Quantos numeros voce vai digitar? ");
		N = sc.nextInt();

	    double[] vet = new double[N];

		for (int i = 0; i < N; i++) {
	    	System.out.print("Digite um numero: ");
			vet[i] = sc.nextDouble();
	    }

		posicao = 0;
		maiorValor = vet[0];

		for (int i = 1; i < N; i++) {
			if (vet[i] > maiorValor) {
				maiorValor = vet[i];
				posicao = i;
			}
		}
		
	    System.out.printf("MAIOR VALOR = %.1f\n", maiorValor);
	    System.out.printf("POSICAO DO MAIOR VALOR = %d\n", posicao);

		sc.close();
	}
}
